### PR TITLE
refactor: cleanup `DateInputButton` & `ExpirationTimePicker`

### LINKED ISSUE #7199: [Task]: Cleanup `DateInputButton` & `ExpirationTimePicker`

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

Closes #7199 

## Changelog

```
- Cleanup `DateInputButton` & `ExpirationTimePicker`
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by marc2332 — APPROVED** [review]

File: packages/shared/components/ExpirationTimePicker.svelte [hunk 1]
```diff
@@ -46,3 +41,12 @@
     </div>
 </button>
 <ExpirationTimePickerMenu bind:this={menu} bind:value bind:selected bind:anchor />
+
+<style lang="scss">
+    button {
+        @apply flex items-center justify-center;
+        &:disabled {
+            @apply cursor-default;
+        }
```
Comment (VmMad) [inline packages/shared/components/ExpirationTimePicker.svelte:50]: not necessary

File: packages/shared/components/ExpirationTimePicker.svelte [hunk 2]
```diff
@@ -4,19 +4,19 @@
     import { formatDate, localize } from '@core/i18n'
     import { TimePeriod } from '@core/utils'
 
-    export let value: Date = undefined
+    export let value: Date | undefined | null = undefined
```
Comment (VmMad) [inline packages/shared/components/ExpirationTimePicker.svelte:7]:
```suggestion
    export let value: Date | null = null
```

**Review by VmMad — CHANGES_REQUESTED** [review]

**Review by VmMad — APPROVED** [review]
Nice

**Comment by marc2332** [comment]
conflicts @evavirseda

**Review by marc2332 — APPROVED** [review]

**Pull request merged**